packages/js/src/cache/notifications-cache.ts

```diff
@@ -0,0 +1,253 @@
+];
+
+export class NotificationsCache {
+  #emitter: NovuEventEmitter;
+  /**
+   * The key is the stringified notifications filter, the values are the paginated notifications.
+   */
+  #cache: Cache<ListNotificationsResponse>;
+
+  constructor() {
+    this.#emitter = NovuEventEmitter.getInstance();
+    notificationEvents.forEach((event) => {
+      this.#emitter.on(event, this.handleNotificationEvent());
+    });
+    notificationEventsThatRemove.forEach((event) => {
+      this.#emitter.on(event, this.handleNotificationEvent({ remove: true }));
+    });
+    notificationsEvents.forEach((event) => {
+      this.#emitter.on(event, this.handleNotificationsEvent());
+    });
+    notificationsEventsThatRemove.forEach((event) => {
+      this.#emitter.on(event, this.handleNotificationsEvent({ remove: true }));
+    });
+    this.#cache = new InMemoryCache();
+  }
+
+  private updateNotificationInCache = (key: string, data: Notification): boolean => {
+    const notificationsResponse = this.#cache.get(key);
+    if (!notificationsResponse) {
+      return false;
+    }
+
+    const index = notificationsResponse.notifications.findIndex((el) => el.id === data.id);
+    if (index === -1) {
+      return false;
+    }
+
+    const updatedNotifications = [...notificationsResponse.notifications];
+    updatedNotifications[index] = data;
+
+    this.#cache.set(key, { ...notificationsResponse, notifications: updatedNotifications });
+
+    return true;
+  };
+
+  private removeNotificationInCache = (key: string, data: Notification): boolean => {
+    const notificationsResponse = this.#cache.get(key);
+    if (!notificationsResponse) {
+      return false;
+    }
+
+    const index = notificationsResponse.notifications.findIndex((el) => el.id === data.id);
+    if (index === -1) {
+      return false;
+    }
+
+    const newNotifications = [...notificationsResponse.notifications];
+    newNotifications.splice(index, 1);
+
+    this.#cache.set(key, {
+      ...notificationsResponse,
+      notifications: newNotifications,
+    });
+
+    return true;
+  };
+
+  private handleNotificationEvent =
+    ({ remove }: { remove: boolean } = { remove: false }) =>
+    ({ data }: { data?: Notification }): void => {
+      if (!data) {
+        return;
+      }
+
+      const uniqueFilterKeys = new Set<string>();
+      this.#cache.keys().forEach((key) => {
+        let isNotificationFound = false;
+        if (remove) {
+          isNotificationFound = this.removeNotificationInCache(key, data);
+        } else {
+          isNotificationFound = this.updateNotificationInCache(key, data);
+        }
+
+        if (isNotificationFound) {
+          uniqueFilterKeys.add(getFilterKey(getFilter(key)));
+        }
+      });
+
+      uniqueFilterKeys.forEach((key) => {
+        const notificationsResponse = this.getAggregated(getFilter(key));
+        this.#emitter.emit('notifications.list.updated', {
+          data: notificationsResponse,
+        });
+      });
```

Comment:
Depending on the received event we decide whether the notification should be removed from the cache or updated in the cache. 
The notification can belong to many different tabs/filters. We must collect unique filters and emit events for each.